enterprise/frontend/src/embedding-sdk/cli/steps/generate-component-files.ts

```diff
@@ -55,22 +70,47 @@ export const generateReactComponentFiles: CliStepMethod = async state => {
   const componentExtension = isInTypeScriptProject ? "tsx" : "jsx";
 
   // Generate sample components files in the specified directory.
-  for (const { name, content } of sampleComponents) {
-    await fs.writeFile(`${path}/${name}.${componentExtension}`, content);
+  for (const { fileName, content } of sampleComponents) {
+    await fs.writeFile(
+      `${reactComponentPath}/${fileName}.${componentExtension}`,
+      content,
+    );
   }
 
   // Generate analytics.css sample styles.
-  await fs.writeFile(`${path}/analytics.css`, ANALYTICS_CSS_SNIPPET);
+  await fs.writeFile(
+    `${reactComponentPath}/analytics.css`,
+    ANALYTICS_CSS_SNIPPET,
+  );
 
   // Generate index.js file with all the component exports.
   const exportIndexContent = sampleComponents
-    .map(file => `export * from "./${file.name}";`)
+    .map(file => `export * from "./${file.fileName}"`)
     .join("\n")
     .trim();
 
-  await fs.writeFile(`${path}/index.${fileExtension}`, exportIndexContent);
+  await fs.writeFile(
+    `${reactComponentPath}/index.${fileExtension}`,
+    exportIndexContent,
+  );
+
+  const hasNextJsCustomAppOrRootLayout =
+    await checkIfNextJsCustomAppOrRootLayoutExists();
+
+  await generateNextJsDemoFiles({
+    hasNextJsCustomAppOrRootLayout,
+    reactComponentPath,
+    componentExtension,
+  });
```

Comment:
Generate the analytics-demo route, as well as the custom app or root layout where needed.